michael-eugene/assets/js/newsapi.js

```diff
@@ -1,79 +1,34 @@
+                        <div><a href="${res.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
+                    </div>
+                </div>
+                `
+            }))
+}
+
 // BBC API
 var url = "https://newsapi.org/v2/top-headlines?sources=bbc-news&apiKey=f3d6aa29b4af4d45ad9cb6542b496c04";
-
-fetch(url)
-    .then((response) => response.json())
-    .then((response) => 
-
-        response.articles.slice(-4).forEach(art => {
-            document.getElementsByClassName('wrapper')[0].innerHTML += `
-        <div class="grid-item sm">
-            <div class="img-wrapper" style="background-image:url(${response.articles[0].urlToImage});"></div>
-            <div class="hero-txt">
-                <h2 class="grid-txt">${response.articles[0].title}</h2>
-                <p class="grid-txt">${response.articles[0].description}</p>
-                <div><a href="${response.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
-            </div>
-        </div>
-        `
-        }))
+fetchAndRenderNews(url, 0);
 
 // Technology API
 var url = "https://newsapi.org/v2/top-headlines?sources=techcrunch&apiKey=f3d6aa29b4af4d45ad9cb6542b496c04";
-
-fetch(url)
-    .then((response) => response.json())
-    .then((response) => 
-
-        response.articles.slice(-4).forEach(art => {
-            document.getElementsByClassName('wrapper')[1].innerHTML += `
-        <div class="grid-item sm">
-        <div class="img-wrapper" style="background-image:url(${response.articles[0].urlToImage});"></div>
-            <div class="hero-txt">
-                <h2 class="grid-txt">${response.articles[0].title}</h2>
-                <p class="grid-txt">${response.articles[0].description}</p>
-                <div><a href="${response.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
-            </div>
-        </div>
-        `
-        }))
+fetchAndRenderNews(url, 1);
 
 // Entertainment API
 var url = "https://newsapi.org/v2/top-headlines?sources=mtv-news&apiKey=f3d6aa29b4af4d45ad9cb6542b496c04";
-
-fetch(url)
-    .then((response) => response.json())
-    .then((response) => 
-
-        response.articles.slice(-4).forEach(art => {
-            document.getElementsByClassName('wrapper')[2].innerHTML += `
-        <div class="grid-item sm">
-        <div class="img-wrapper" style="background-image:url(${response.articles[0].urlToImage});"></div>
-            <div class="hero-txt">
-                <h2 class="grid-txt">${response.articles[0].title}</h2>
-                <p class="grid-txt">${response.articles[0].description}</p>
-                <div><a href="${response.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
-            </div>
-        </div>
-        `
-        }))
+fetchAndRenderNews(url, 2);
 
 // Sport API
 var url = "https://newsapi.org/v2/top-headlines?sources=espn&apiKey=f3d6aa29b4af4d45ad9cb6542b496c04";
```

Comment:
Nice work on refactoring. You have many `url` variables. You could rename them. Like this can be `sportUrl`

Reply:
I'll do that once I get power on my laptop